cupyx/scipy/interpolate/_interpolate.py

```diff
@@ -0,0 +1,986 @@
+            remainder_int = cupy.empty_like(range_int)
+            if b <= xe:
+                _integrate(
+                    self.c.reshape(self.c.shape[0], self.c.shape[1], -1),
+                    self.x, a, b, False, out=remainder_int)
+                range_int += remainder_int
+            else:
+                _integrate(
+                    self.c.reshape(self.c.shape[0], self.c.shape[1], -1),
+                    self.x, a, xe, False, out=remainder_int)
+                range_int += remainder_int
+
+                _integrate(
+                    self.c.reshape(self.c.shape[0], self.c.shape[1], -1),
+                    self.x, xs, xs + left + a - xe, False, out=remainder_int)
+                range_int += remainder_int
+        else:
+            _integrate(
+                self.c.reshape(self.c.shape[0], self.c.shape[1], -1),
+                self.x, a, b, bool(extrapolate), out=range_int)
+
+        # Return
+        range_int *= sign
+        return range_int.reshape(self.c.shape[2:])
+
+    def solve(self, y=0., discontinuity=True, extrapolate=None):
+        """
+        Find real solutions of the equation ``pp(x) == y``.
+
+        Parameters
+        ----------
+        y : float, optional
+            Right-hand side. Default is zero.
+        discontinuity : bool, optional
+            Whether to report sign changes across discontinuities at
+            breakpoints as roots.
+        extrapolate : {bool, 'periodic', None}, optional
+            If bool, determines whether to return roots from the polynomial
+            extrapolated based on first and last intervals, 'periodic' works
+            the same as False. If None (default), use `self.extrapolate`.
+
+        Returns
+        -------
+        roots : ndarray
+            Roots of the polynomial(s).
+            If the PPoly object describes multiple polynomials, the
+            return value is an object array whose each element is an
+            ndarray containing the roots.
+
+        Notes
+        -----
+        This routine works only on real-valued polynomials.
+        If the piecewise polynomial contains sections that are
+        identically zero, the root list will contain the start point
+        of the corresponding interval, followed by a ``nan`` value.
+        If the polynomial is discontinuous across a breakpoint, and
+        there is a sign change across the breakpoint, this is reported
+        if the `discont` parameter is True.
+
+        At the moment, this routine will call the SciPy (CPU)
+        implementation.
+        """
+        if not scipy_found:
+            raise NotImplementedError(
+                'At the moment there is not a GPU implementation for '
+                'solve and SciPy was not found')
+
+        cpu_pp = ScPPoly(self.c.get(), self.x.get(),
+                         self.extrapolate, self.axis)
+        solution = cpu_pp.solve(y, discontinuity, extrapolate)
```

Comment:
Since https://github.com/cupy/cupy/issues/6359 has no current solution and `solve` depends on eigenvalue solve, we decided to delegate the implementation to the SciPy one, which means that the coefficients and the breakpoints are copied to CPU. Please let me know if you'd prefer to omit this method altogether

Reply:
Thanks for the suggestion, but I'd prefer to let this method just raise `NotImplementedError` (unconditionally if scipy is there or not) to avoid data transfer and computation on CPU that users can not appearently find.